### ISSUE TITLE
BATS reports success despite failing test cases, regression from v1.5.0 to v1.6.0

### ISSUE THREAD

**Issue by Richard-Neill** [bats-core/bats-core#575]
**Describe the bug**
In v1.6.0, BATS can return success despite failing test cases. The exit status of a BATS 'run' command called from the teardown_file() function is returned as the exit status of the bats script. This means that even if the test cases fail, a successful teardown will produce a passing test suite execution.

This appears to be a regression from v1.5.0, where the behaviour is as expected (see below)

**To Reproduce**

Minimal example `tests.bats` file:

```
#!/usr/bin/env bats

my_teardown_func() {
     return 0 # return code of this function controls the exit status of the bats execution
}

teardown_file() {
    run my_teardown_func
}

@test 'my_test' {
    return 1 
}
```

Running this on each version:

v1.6.0:
```
user@machine:~/$ bats --version
Bats 1.6.0
user@machine:~/$ bats tests.bats
 ✗ my_test
   (in test file tests.bats, line 14)
     `return 1' failed

1 test, 1 failure

user@machine:~/$ echo $?
0
```

v1.5.0:
```
user@machine:~/$ bats --version
Bats 1.5.0
user@machine:~/$ bats tests.bats
 ✗ my_test
   (in test file tests.bats, line 14)
     `return 1' failed

1 test, 1 failure

user@machine:~/$ echo $?
1
```

**Expected behavior**
I expected the behaviour to be the same: exit status of `bats tests.bats` for both v1.6.0 and v1.5.0 would be a failure, due to the failing `my_test` test case.

**Environment (please complete the following information):**
 - Bats Version v1.6.0 tag (210acf3)
 - OS: 5.14.0-1031-oem Ubuntu SMP Fri Mar 25 08:21:05 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
 - Bash version: GNU bash, version 5.0.17(1)-release (x86_64-pc-linux-gnu)

**Comment by martin-schulze-vireso** [comment]
I quick bisect found this commit to be the culprit: 82c11fb57126aad18949f24c64b6046c164c53b9
It re/moved a (seemingly) unnecessary declaration of `local status`.

Also, I am curious why you are using `run` in a `teardown` function? I have always operated under the assumption that `run` is only used inside `@test` functions.

**Comment by Richard-Neill** [comment]
@martin-schulze-vireso I wonder, would this fix be considered for backporting to the latest release?

**Comment by martin-schulze-vireso** [comment]
You mean a 1.6.1? I am currently working on the 1.7.0 release to come out soonish.

**Comment by Richard-Neill** [comment]
Yeah I was thinking of a 1.6.1. I'm not aware of your release process so I just thought I'd ask.

For context, I work using the Yocto Project and meta-openembedded which builds the latest release (therefore 1.6.0). Perhaps I can apply 5f372058b05f817e4e3a8dab27f83c30fd467504 locally onto 1.6.0 as an alternative.

**Comment by martin-schulze-vireso** [comment]
I am already pretty strained so my plan is to get 1.7.0 out with these fixes asap without detours. Doing a backport would take even more time and would have to go farther back than 1.6.0.

**Comment by martin-schulze-vireso** [comment]
@Richard-Neill  FYI, I just released 1.6.1 after getting another report of the same issue. It seems I underestimated how many people are using run in `teardown_file`.
